File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.java
// Last updated: 8/4/2026, 12:40:54 PM
class Solution {
    public int[] buildArray(int[] nums) {
        int arr[] = new int[nums.length];
        for(int i = 0 ; i < nums.length ; i++){
            arr[i] = nums[nums[i]];
        }
        return arr;
    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.cpp
// Last updated: 8/4/2026, 12:41:03 PM
class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        vector<int>ans;
        for(int i=0;i<nums.size();i++){
            ans.push_back(nums[nums[i]]);
        }
        return ans;
    }
};